### DISCUSSION TITLE
Allow duplicating blocks within matrix field, when blocks are displayed as cards

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#15773] [Ideas] [answered]
I usually use cards view within the matrix fields. Element index view mode offers too much clutter i usually don't use there, such as filters and search from, export buttons and so on.
However, there is one functionality i find very useful in the element index view - duplicating blocks.

Would it be possible to add duplication option to card view within a matrix field?

**Comment by ryanleichty** [comment]
We're migrating from Neo and would also appreciate this functionality 🙏

**Comment by Nellyaa** [comment]
+1 here! Duplicating and Copy/Paste-Blocks would be great.

**Comment by alexanderbuergin** [comment]
Yes Please.

As well as **Duplicating** and **Copy/Paste**!

Make this Option's Core:
https://plugins.craftcms.com/smith
https://plugins.craftcms.com/matrix-extended?craft5

See https://github.com/craftcms/cms/discussions/15895 as well

**Comment by garycrane** [comment]
**+1 here too**. A little piece of me dies every time I have to manually recreate a row which is _almost identical_ to another row on the page, or if I need to recreate exactly the same row on another page. Happens all the time.  _Pleeeeaaaase_ bring Matrix row duplication and copy/paste into core 🙏🙏🙏 The two plugins that are currently available don't really do the job perfectly, especially where cards are concerned.

**Reply by alexanderbuergin** [reply]
@brandonkelly Please. 🙏🙏🙏 😊

**Comment by garycrane** [comment]
Some good news! The developer of Matrix Extended (@vandres) has been working on a cards integration 🥳
There's a dev-cards branch available that you can try. It has a couple of known bugs that Volker is working through, but in my tests it's looking pretty good so far!

[Here are the instructions on how to install and get started.](https://github.com/vandres/craft-matrix-extended/issues/12#issuecomment-2505779427)

**Comment by brandonkelly** [comment, marked as answer]
Nested element duplication is coming in Craft 5.7 :tada: (#16819)

**Reply by brandonkelly** [reply]
Craft 5.7.0 is out with nested element duplication 🎉

**Comment by MangoMarcus** [comment]
@brandonkelly ~~Do you think duplicated elements should be added after the original block? That's the behaviour in the matrix-extended plugin and I believe in the Smith plugin. At the moment they're added at the end of all blocks.~~

Edit: does actually seem a bit hit and miss, may be a compatibility problem with a 3rd party plugin

**Reply by alexanderbuergin** [reply]
I would love to see that one.

**Reply by garycrane** [reply]
For me, duplicated blocks always appear directly beneath the original block. However when copying, I only get the option to paste at the bottom. Having the option to paste between any row would be nice.

**Reply by MangoMarcus** [reply]
Sorry I've done some more testing and it does seem to be hit and miss, but more hit than miss. I'm wondering if it's compatibility with the matrix-extended plugin so feel free to ignore this.

+1 for pasting between blocks though

**Reply by mmikkel** [reply]
https://github.com/craftcms/cms/discussions/17406

**Reply by garycrane** [reply]
> I'm wondering if it's compatibility with the matrix-extended plugin so feel free to ignore this.

Yeah, I think Volker has removed loads of the duplicating/copying functionality from v5 of Matrix Extended because it was buggy. Not really needed now that it's native 😀
